zerver/models.py

```diff
@@ -403,7 +407,26 @@ def get_realm_emoji_uncached(realm: Realm) -> Dict[Text, Dict[str, Any]]:
                 'id': row.author.id,
                 'email': row.author.email,
                 'full_name': row.author.full_name}
-        d[row.name] = dict(id=row.id,
+        d[str(row.id)] = dict(id=str(row.id),
+                              name=row.name,
+                              source_url=get_emoji_url(row.file_name, row.realm_id),
+                              deactivated=row.deactivated,
+                              author=author)
+    return d
+
+def get_active_realm_emoji_uncached(realm: Realm) -> Dict[str, Dict[str, Any]]:
+    d = {}
+    from zerver.lib.emoji import get_emoji_url
+    for row in RealmEmoji.objects.filter(realm=realm,
+                                         deactivated=False).select_related('author'):
+        author = None
+        if row.author:
+            author = {
+                'id': row.author.id,
+                'email': row.author.email,
+                'full_name': row.author.full_name}
+        d[row.name] = dict(id=str(row.id),
+                           name=row.name,
```

Comment:
I think the refactor to create `get_active_realm_emoji` and use it in Bugdown might be something you can extract as a prep commit that we can merge early; it doesn't seem to depend on the change in addressing.  I think that would help make it easy to merge the rest of this.

Reply:
Okay will get it done!